subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -668,9 +671,16 @@ static int location_process(const char * const buf)
 {
 #if defined(CONFIG_NRF_CLOUD_LOCATION) && defined(CONFIG_NRF_CLOUD_MQTT)
 	if (location_cb) {
-		struct nrf_cloud_location_result res;
-		int ret = nrf_cloud_location_process(buf, &res);
 
+		int ret;
+		struct nrf_cloud_location_result res = {0};
+
+#if defined(CONFIG_NRF_CLOUD_LOCATION_ANCHOR_LIST)
+		res.anchor_buf = anchor_list_buf;
+		res.anchor_buf_sz = sizeof(anchor_list_buf);
+#endif
```

Comment:
One potential issue: What happens when we get a new location result while the user is still processing the old one?

We should probably include an explicit warning in the struct comments and the KConfig comments that the provided location buffer is ONLY valid during the callback, and that any long-term use should be done by copying the results somewhere else.

And then I think we should also use a semaphore to protect this buffer while `location_cb` is being evaluated

And then, finally, we should probably wipe the buffer immediately after `location_cb` is called just to help enforce this.

Reply:
this is done in the context of the mqtt event handler

Reply:
I'm more just fearful that the end-user attempts to retain a copy of what they were given. But you're right, the semaphore isn't needed. But explicit warnings not do that / some mild enforcement by wiping the static buffer both seem potentially wise